modules/puzzle/src/main/PuzzleFinisher.scala

```diff
@@ -199,18 +199,16 @@ final private[puzzle] class PuzzleFinisher(
       if player.clueless then glicko._1
       else glicko._1.average(glicko._2, weightOf(angle, win))
 
-  private val VOLATILITY = lila.rating.Glicko.default.volatility
-  private val TAU        = 0.75d
-  private val calculator = glicko2.RatingCalculator(VOLATILITY, TAU)
+  private val calculator = lila.rating.Glicko.system
```

Comment:
what the fudge... there was a horrible bug here.

```scala
private val calculator = glicko2.RatingCalculator(VOLATILITY, TAU)
```
was completely wrong here, as the constructor goes `RatingCalculator(tau: Double = 0.75d, ratingPeriodsPerDay: Double = 0)`.

And that's what happens when we use weak types like `Double`. Arguments get swapped during some code change and no-one notices.

Gonna change these to proper `opaque type` so that it doesn't happen again.

Reply:
Goodness, that is quite a mess and I feel somewhat bad for having introduced it years ago.

Parameter `tau` being near-zero may have caused solvers' volatility to increase slower when on a hot streak:
> Smaller values of τ prevent the volatility measures from changing by large amounts, which in turn prevent enormous changes in ratings based on very improbable results.
> http://www.glicko.net/glicko/glicko2.pdf

Parameter `ratingPeriodsPerDay` being `0;75d` instead of `0.21436d` may have caused solvers' RD to increase faster.

Reply:
Note that I wasn't blaming you for the bug. In fact I didn't check who introduced it and assumed it was me.
Thanks for fixing it tho.